packages/target-electron/src/windows/main.ts

```diff
@@ -164,21 +159,11 @@ export function init(options: { hidden: boolean }) {
     }
   }
   window.webContents.session.setPermissionCheckHandler((_wc, permission) => {
-    if (systemPreferences.getMediaAccessStatus && permission === 'media') {
-      return systemPreferences.getMediaAccessStatus('camera') === 'granted'
-    }
-    // if (systemPreferences.getMediaAccessStatus && permission === "microphone") {
-    //   return systemPreferences.getMediaAccessStatus("microphone") === "granted"
-    // }
     return permission_handler(permission as any)
   })
   window.webContents.session.setPermissionRequestHandler(
     (_wc, permission, callback) => {
-      if (systemPreferences.askForMediaAccess && permission === 'media') {
-        systemPreferences.askForMediaAccess('camera').then(callback)
-      } else {
-        callback(permission_handler(permission))
-      }
+      callback(permission_handler(permission))
```

Comment:
I'm not sure what this change is about, but does the QR scanner still work on a fresh instance of DC on iOS? Now we don't seem to ever request camera permission.

Reply:
The camera is requested on the call
navigator.mediaDevices.getUserMedia({
       video: videoConstraints,
       audio: false,
 })
 
The mernissionHandler can not distinguish between camera access & microphone access so if you block camera access you also block microphone access - so it is not usable

Reply:
Huh, but microphone access does _not_ get requested on

https://github.com/deltachat/deltachat-desktop/blob/d887c6f078158fd91c7fc8386a38e1d511a83678/packages/frontend/src/components/AudioRecorder/MicRecorder.ts#L131

?

Reply:
There the user would be asked to access the microphone on first start if we didn't call askForMediaAccess('microphone') before. If he denies there he will be shown a message "Permission denied" and so any next time he tries to use the microphone.
Our permissionHandler would not help, since it can't distinguish between mic or camera access, it's just the "media" permission.

If the camera access is denied in QrReader we show always the explanation about the permission to access the camera instead of the video so the user is informed there. And we did ignore the case that the media permission could be for audio since it was only needed for camera so far.

Another question would be, how to disable the mic and/or camera access on Linux. Maybe we could implement an own handling for that?